### ISSUE TITLE
ENH: Automatic conversion of variables referring to time periods

### ISSUE THREAD

**Issue by hmgaudecker** [iza-institute-of-labor-economics/gettsim#575]
#### Is your feature request related to a problem?

We currently have lots of `x * 12`, `z / 12`, etc. inside of functions to convert between time units. This is error-prone and adds unnecessary complexity to functions.

#### Describe the solution you would like to see

- Rename all yearly variables ([GEP 1](https://gettsim.readthedocs.io/en/latest/geps/gep-01.html#column-names-a-k-a-variables-in-stata)) from `xxx` to `xxx_y` or `zzz_hh` to `zzz_y_hh`.
  - Most functions in taxes
  - ~~Some input variables~~
- Provide conversion functions similar to unit aggregation functions for `_hh`, `_tu`:
  - call them `m_to_y`, `y_to_d`, etc.
  - Regex for variables ending in `_[ymwd]` or `_[ymwd]_(hh|tu)` (not tested)
  - Say we have a variable `xxx_y` and `xxx_m` is requested. We then automatically add a function to the DAG like `lambda z: z / 12`
  - Based on yearly values, conversion factors should be 12 - 365.25/7 - 365.25
  - This ensures that round-trips are possible
  - As with aggregation functions, if there is an explicit function for `xxx_m` (in the example above), no automatic conversion is added.
- Search for `[\*\/] ?12` in `src/_gettsim` and replace
  - input variables by the required temporal level
  - explicit calculations by function calls to, say, `y_to_m` if the calculation relates to an entry in the params dict etc.
  - sometimes the level of the calculation (e.g., `_lohnst_m` should not exist anymore as an explicit function, replace by `_lohnst_y`)

**Comment by lars-reimann** [comment]
Should this also get the `_y` suffix?

https://github.com/iza-institute-of-labor-economics/gettsim/blob/6f58ccf617ea1e4601dfd1c61c9fb46cfb353224/src/_gettsim/taxes/zu_verst_eink/eink.py#L20

**Comment by hmgaudecker** [comment]
Yes, I think everything in that module that does not already have an `_m` suffix, except for `rente_ertragsanteil`.

**Comment by hmgaudecker** [comment]
Sure! Excluding is much (!) easier:

```
gettsim/src/_gettsim/taxes/eink_st.py
  63,1: def _eink_st_tarif(x: float, params: dict) -> float:
  141,1: def kinderfreib_günstiger_tu(
```

**Comment by hmgaudecker** [comment]
I also skimmed the input variables and don't think anything should be affected there, crossed that out above.

**Comment by lars-reimann** [comment]
How should this function be called?

https://github.com/iza-institute-of-labor-economics/gettsim/blob/818fe72760835307eb1a16f46f6132ee6fce6dcf/src/_gettsim/taxes/zu_verst_eink/vorsorgeaufw.py#L75

Where `_tu` is used as a suffix, I've replaced it by `y_tu` (e.g. `alleinerz_freib_tu` -> `alleinerz_freib_y_tu`). Here it's in the middle of the name, however.

**Comment by hmgaudecker** [comment]
`einführung_vorsorgeaufw_y_tu`, made a mistake there during the dates active introduction, I believe. Please change all occurrences, `_tu` and`_hh` should always be suffixes after applying the decorator.

**Comment by mjbloemer** [comment]
> We currently have lots of `x * 12`, `z / 12`, etc. inside of functions to convert between time units.

Sometimes the law has phrases like "... ein Zwölftel des steuerlichen ..." or "... das zwölffache der Bemessungsgrundlage nach § ..." (both invented phrases, but I could look up real examples), so it would be natural to transform the law to code exactly like this.

**Comment by hmgaudecker** [comment]
But that is precisely what the `_m` / `_y` conversion does? In the end it remains a question of readability. 

And yes, there may indeed be cases where one might want to keep the explicit thing in the code (avoiding `eink_st_m` might be such a case).
